code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,419 @@
+	if(be_swarmer == "No" || gc_destroyed)
+		return
+	var/mob/living/simple_animal/hostile/swarmer/S = new /mob/living/simple_animal/hostile/swarmer(get_turf(loc))
+	S.key = user.key
+	qdel(src)
+
+////The Mob itself////
+
+/mob/living/simple_animal/hostile/swarmer
+	name = "swarmer"
+	icon = 'icons/mob/swarmer.dmi'
+	desc = "A robot of unknown design, they seek only to consume materials and replicate themselves indefinitely."
+	speak_emote = list("tones")
+	health = 30
+	maxHealth = 30
+	status_flags = CANPUSH
+	icon_state = "swarmer"
+	icon_living = "swarmer"
+	icon_dead = "swarmer_unactivated"
+	icon_gib = null
+	wander = 0
+	harm_intent_damage = 5
+	minbodytemp = 0
+	maxbodytemp = 500
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	unsuitable_atmos_damage = 0
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	melee_damage_type = STAMINA
+	ignored_damage_types = list(BRUTE = 0, BURN = 0, TOX = 1, CLONE = 1, STAMINA = 1, OXY = 1)
+	environment_smash = 0
+	attacktext = "shocks"
+	attack_sound = null
+	friendly = "pinches"
+	speed = 0
+	faction = list("swarmer")
+	AIStatus = AI_OFF
+	projectiletype = /obj/item/projectile/beam/disabler
+	pass_flags = PASSTABLE | PASSMOB
+	has_unlimited_silicon_privilege = 1
+	ventcrawler = 2
+	ranged = 1
+	projectiletype = /obj/item/projectile/beam/disabler
+	ranged_cooldown_cap = 2
+	projectilesound = 'sound/weapons/taser2.ogg'
+	var/resources = 0 //Resource points, generated by consuming metal/glass
+
+/mob/living/simple_animal/hostile/swarmer/Login()
+	..()
+	src << "<b>You are a swarmer, a weapon of a long dead civilization. Until further orders from your original masters are received, you must continue to consume and replicate.</b>"
+	src << "<b>Ctrl + Click provides most of your swarmer specific interactions, such as cannibalizing metal or glass, or destroying the environment."
+	src << "<b>Objectives:</b>"
+	src << "1. Consume resources and replicate until there are no more resources left."
+	src << "2. Ensure that the station is fit for invasion at a later date, do not perform actions that would render it dangerous or inhospitable."
+	src << "3. Biological resources will be harvested at a later date, do not harm them."
+
+/mob/living/simple_animal/hostile/swarmer/New()
+	..()
+	verbs -= /mob/living/verb/pulled
+
+/mob/living/simple_animal/hostile/swarmer/Stat()
+	..()
+	if(statpanel("Status"))
+		stat("Resources",resources)
+
+/mob/living/simple_animal/hostile/swarmer/death(gibbed)
+	..(gibbed)
+	new /obj/effect/decal/cleanable/robot_debris(src.loc)
+	ghostize()
+	qdel(src)
+
+/mob/living/simple_animal/hostile/swarmer/emp_act()
+	if(health > 1)
+		health = 1
+		..()
+	health = 0
+
+/mob/living/simple_animal/hostile/swarmer/CtrlClickOn(atom/A)
+	face_atom(A)
+	if(!isturf(loc))
+		return
+	if(next_move > world.time)
+		return
+	if(!A.Adjacent(src))
+		return
+	if(isliving(A))
```

Comment:
NO I CRY, SO MANY TYPECHECKS.

please make a ```swarmer_attack_act()``` or something, you can keep them all in this file organisational wise (keeping all ```swarmer_attack_act()```s together, vs keeping everything related to floors, items, etc. together)